ts/components/helpers/withErrorModal.tsx

```diff
@@ -0,0 +1,161 @@
+import { Option } from "fp-ts/lib/Option";
+import { Button, H2, Text } from "native-base";
+import * as React from "react";
+import { Image, Modal, StyleSheet, View } from "react-native";
+import { connect } from "react-redux";
+
+import I18n from "../../i18n";
+import { Action, ReduxProps } from "../../store/actions/types";
+import { GlobalState } from "../../store/reducers/types";
+import variables from "../../theme/variables";
+
+export type WithErrorModalInjectedProps = {
+  error: Option<string>;
+};
+
+export type WithErrorModalProps<P> = P &
+  WithErrorModalInjectedProps &
+  ReduxProps;
+
+const styles = StyleSheet.create({
+  contentWrapper: {
+    flex: 1,
+    padding: variables.contentPadding
+  },
+
+  imageAndMessageContainer: {
+    flex: 1,
+    justifyContent: "center",
+    alignItems: "center"
+  },
+
+  image: {
+    marginBottom: 20
+  },
+
+  buttonsContainer: {
+    flexDirection: "row",
+    marginTop: "auto"
+  },
+
+  buttonCancel: {
+    flex: 4,
+    backgroundColor: variables.brandDarkGray
+  },
+
+  buttonCancelText: {
+    color: variables.colorWhite
+  },
+
+  separator: {
+    width: 10
+  },
+
+  buttonRetry: {
+    flex: 8
+  }
+});
+
+/**
+ * A HOC to display a modal to notify the user an error occurred.
+ * Inside the modal cancel and retry buttons are rendered conditionally.
+ *
+ * @param WrappedComponent The react component you want to wrap
+ * @param errorSelector A redux selector that returns the error (as string) or undefined
+ * @param errorMessage The error message to show when errorSelector returns a value
+ * @param cancelActionCreator A function that accept the WrappedComponent props and return a redux Action to dispatch.
+ *                            We pass in props to make possible to create action like cancelSomething(props.id)
+ * @param retryActionCreator A function that accept the WrappedComponent props and return a redux Action to dispatch.
+ *                           We pass in props to make possible to create action like loadSomething(props.id)
+ */
+export function withErrorModal<P>(
+  WrappedComponent: React.ComponentType<P>,
+  errorSelector: (state: GlobalState) => Option<string>,
+  errorMessage: string,
+  cancelActionCreator: (
```

Comment:
what's the benefit of having this component dispatch the action? can't we just make it accept a simple `() => void` function (like we do with _goBack_?) I don't see the need to making this component aware of redux/dispatch/etc...